packages/compass-query-bar/src/components/query-history/recent-list.tsx

```diff
@@ -27,31 +29,52 @@ type RecentActions = {
   onFavorite: (query: RecentQuery, name: string) => Promise<boolean>;
   onDelete: (id: string) => void;
   onApply: (query: BaseQuery) => void;
+  onUpdateRecentChoosen: () => void;
 };
 
 const RecentItem = ({
   query,
+  isReadonly,
   onFavorite,
   onDelete,
   onApply,
+  onUpdateRecentChoosen,
 }: RecentActions & {
   query: RecentQuery;
+  isReadonly: boolean;
 }) => {
+  const readOnlyCompass = usePreference('readOnly', React);
+  const isUpdateQuery = !!query.update;
+  const isDisabled = isUpdateQuery && (isReadonly || readOnlyCompass);
+
   const formRef = React.useRef<HTMLFormElement>(null);
   const [isAddingFavorite, setIsAddingFavorite] = useState(false);
   const attributes = useMemo(() => getQueryAttributes(query), [query]);
   const title = useFormattedDate(query._lastExecuted.getTime());
 
+  const onClickRecent = useCallback(() => {
+    if (isDisabled) {
+      return;
+    }
+
+    if (isUpdateQuery) {
+      onUpdateRecentChoosen();
+    }
+
+    track('Query History Recent Used');
```

Comment:
Would we want to distinguish via an event property between using using a recent as a find query vs. a bulk operation?

Reply:
Probably we should, there is a task for telemetry, but it's still not pretty well defined. We are maintaining the behaviour we have for find queries.